rustls/src/server/hs.rs

```diff
@@ -30,14 +30,14 @@ pub(super) type NextState = Box<dyn State<ServerConnectionData>>;
 pub(super) type NextStateOrError = Result<NextState, Error>;
 pub(super) type ServerContext<'a> = crate::conn::Context<'a, ServerConnectionData>;
 
-pub(super) fn incompatible(common: &mut CommonState, why: &str) -> Error {
+pub(super) fn incompatible(common: &mut CommonState, why: PeerIncompatible) -> Error {
     common.send_fatal_alert(AlertDescription::HandshakeFailure);
-    Error::PeerIncompatibleError(why.to_string())
+    Error::PeerIncompatible(why)
```

Comment:
Want tot use `why.into()` here instead (also in `bad_version()`)?

Reply:
I find `thing.into()` is a clarity vs verboseness tradeoff. Here we have `incompatible`, `bad_version`, `decode_error` functions already providing the verboseness saving, so there's no need to make them less clear (and force the reader to go and find the various `From<PeerIncompatible>`, `Into<Error>` that could be in play).

Reply:
Fair enough.